AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -5468,6 +5470,313 @@ function New-AzDnsForwarder {
+    a share on the cloud if not already present.
+    .OUTPUTS
+    PSStorageAccount, representing Storage account provided if exists, else null.
+    #>
+
+    param(
+        [Parameter(Mandatory=$true, Position=0)]
+        [string]$ResourceGroupName,
+
+        [Parameter(Mandatory=$true, Position=1)]
+        [string]$StorageAccountName,
+
+        [Parameter(Mandatory = $true, Position=2)]
+        [string]$Destinationshare
+    )
+
+    Process
+    {
+        Write-Verbose -Message "Getting details of Storage Account:$StorageAccountName."
+
+        $storageAccount = Get-AzStorageAccount -ResourceGroupName $ResourceGroupName -Name $StorageAccountName
+
+        if($storageAccount -ne $null)
+        {
+            Write-Verbose -Message "Checking if the Storage Account is AD joined."
+
+            if($storageAccount.AzureFilesIdentityBasedAuth.DirectoryServiceOptions -ceq 'AD')
+            {
+                Write-Verbose -Message "Storage account is AD joined - Proceeding to check for destination share."
+
+                $accountKeys= Get-AzStorageAccountKey -ResourceGroupName $ResourceGroupName -Name $StorageAccountName | Where-Object {$_.KeyName -like "key1"}
+                $storageAccountContext = New-AzStorageContext -StorageAccountName $StorageAccountName -StorageAccountKey $accountKeys.Value
+
+                Write-Verbose -Message "Checking if the destination share exists"
+
+                $newshare = Get-AzStorageShare -Context $storageAccountContext -Name $DestinationShareName -Erroraction 'silentlycontinue'
+
+                # If the destination share does not exist, the following step will create a new share.
+                if($newshare -eq $null)
+                {
+                    Write-Verbose -Message  "The Destination Share doesn't exist. Creating a new share with the name provided by the user"
+                    New-AzStorageShare -Name $DestinationShareName -Context $storageAccountContext
+                }
+            }
+            #If SA is not AD connect, the user has to connect the SA to AD using the steps in the URL mentioned.
+            else
+            {
+                Write-Error -Message "To Proceed, you need to have Storage Account connected to Active Directory.
+                                        Refer to - https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-auth-active-directory-enable"
+                            -Erroraction Stop
+            }
+        }
+        else
+        {
+            Write-Error -Message "The Storage Account doesn't exist. To create the Storage account and connect it to active directory, 
+                                    please follow the link https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-auth-active-directory-enable" 
+                        -ErrorAction Stop
+        }
+
+        return $storageAccount
+    }
+}
+
+function ACLMigration 

```

Comment:
Need a better name, something like "Migrate-ShareAclToAzure"? Basically the name needs to conform to the powershell "Verb-Noun" convention and also be descriptive about what it is going to do.

Reply:
Followed Verb-Noun convestion. Changed the name